src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -215,14 +216,28 @@ async function getCrossFileCandidates(editor: vscode.TextEditor): Promise<string
      * 2. has the same file extension or it's one of the dialect of target file (e.g .js vs. .jsx)
      * 3. is not a test file
      */
-    return await getOpenFilesInWindow(async candidateFile => {
+    const unsortedCandidates = await getOpenFilesInWindow(async candidateFile => {
         return (
             targetFile !== candidateFile &&
             (path.extname(targetFile) === path.extname(candidateFile) ||
                 (dialects && dialects.has(path.extname(candidateFile)))) &&
             !(await isTestFile(candidateFile, { languageId: language }))
         )
     })
+
+    return unsortedCandidates
+        .map(candidate => {
+            return {
+                file: candidate,
+                fileDistance: getFileDistance(targetFile, candidate),
+            }
+        })
+        .sort((file1, file2) => {
+            return file1.fileDistance - file2.fileDistance
+        })
+        .map(fileToDistance => {
+            return fileToDistance.file
+        })
```

Comment:
nit: the mappings seem to be unnecessary? we can do `sort((file1, file2) => { return  getFileDistance(targetFile, file1) - getFileDistance(targetFile, file2) }`?

Reply:
Would calling `getFileDistance` in sort() would incur redundant calculations?

Reply:
I see, map makes sense if `getFileDistance` does heavy calculations